### ISSUE TITLE
support for this repo in Azure Government Cloud?

### ISSUE THREAD

**Issue by delbert** [Azure-Samples/azure-search-openai-demo#1475]
hello Everyone,

does anyone know if there is an existing fork of this repo that runs in Azure Government Cloud?

if not, does anyone know of any issues with the underlying services that would prevent me from doing such a port, as long as i change over the referenced endpoints and Client mechanisms, such as using: 

`SearchEndpointClient( endpoint=, credential=, audience=AzureAuthorityHosts.AZURE_GOVERNMENT)`

thanks in advance for any help or pointers

**Comment by danieljurek** [comment]
@delbert -- No fork necessary. Well-known sovereign clouds are already supported. https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/sovereign-clouds 

Please give it a try and let us know if you have any feedback!

**Comment by delbert** [comment]
> ~@delbert -- No fork necessary. Well-known sovereign clouds are already supported. https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/sovereign-clouds~
> 
> ~Please give it a try and let us know if you have any feedback!~
> 
> I misread the context on this issue. Let me take a look at this specific template.

thanks, @danieljurek!  i am specifically looking at code in prepdocs.py, searchmanager.py, and strategy.py -- trying to ingest some sample docs with a Azure OpenAI, AI Search, blob storage, all in Gov Cloud

**Comment by pamelafox** [comment]
I am not aware of issues, but am not positive if it has been deployed to the Government Cloud. I did look into removing the hard-coded URLs in our Bicep files, but ran into an issue where not all government URLs were available programmatically: https://github.com/Azure/bicep/issues/12482
But that shouldn't affect you since you know you only want it working for Cloud. Please report back if you try it and discover any issues.

**Comment by delbert** [comment]
thanks, @pamelafox.  curiouser and curiouser.  you start pulling on a thread and its amazing what comes out of the tapestry at you.  
we will definitely report back what we find.  we have started by manually deploying the services via the portal in gov cloud (with private endpoints and policy and rbac restrictions in place) and then are running the prepdocs sequence and deploying the chat app.  hopin for a chat-on-data solution relatively quickly.
we are also trying the template as is, in a gov cloud sandbox subscription with no restrictions on public ip, policy, and rbac to see how that goes.

**Comment by pamelafox** [comment]
There's also the integrated vectorization approach that you can enable in this repository, versus running the local script. The search team says that Integrated Vectorization is compatible with government cloud, as long as you have an OpenAI in that cloud.

**Comment by delbert** [comment]
thanks! super-helpful, at this point.  i'll give it a try

**Comment by mattgotteiner** [comment]
I am wondering if the hardcoded DNS zones (e.g. search.windows.net) would interfere with deploying to a different cloud

**Comment by delbert** [comment]
> thanks! super-helpful, at this point. i'll give it a try

unfortunately, [this limitation](https://learn.microsoft.com/en-us/azure/search/vector-search-integrated-vectorization#limitations) will disqualify Integrated Vectorization for almost any Gov Cloud implementation:  

[Shared private link connections](https://learn.microsoft.com/en-us/azure/search/search-indexer-howto-access-private) to a vectorizer

**Comment by delbert** [comment]
> I am wondering if the hardcoded DNS zones (e.g. search.windows.net) would interfere with deploying to a different cloud

yes it would.  in my case, though, the customer uses their own DNS service, not Azure's

**Comment by pamelafox** [comment]
@delbert Were you ever able to make progress on this? We've had other developers inquire.

**Comment by khallissey** [comment]
@delbert  @pamelafox Has there been any updates on this? Was trying to do this myself too but ran into issues.

**Comment by khallissey** [comment]
But you did get it working using that? Just trying to figure out if it is worth pursuing. Thank you!

**Comment by delbert** [comment]
yes, our solution is working.  sorry if i didn't make that clear

**Comment by khallissey** [comment]
Would you be willing to share what kind of code changes you wound up making?

**Comment by delbert** [comment]
i will share what i can.  there were a number of spots that our implementation had varied from the basic accelerator, such as adding multiple search indeces and switching between them based on the context of the question and using certain models for certain parts of the formulation of the response.  we also built Terraform templates to deploy the Azure Services necessary for the solution and the AZ developer CLI to deploy the web app.  in the web app and in the prepdocs.py we made the following changes:
- we changed all of the endpoints to the Gov Cloud equivalent
- used keys instead of EntraID Auth(at the time PY identity libraries not in Gov)
- used Key Vault so we could rotate keys
- made some changes to fit the app within the customer's Azure Policies

i hope this helps!

**Comment by bcperry** [comment]
@pamelafox, is there appetite for a PR on this?  I was also interested in making this work in the gov cloud and have been able to deploy and test it in an internal gov subscription.  I was also able to parameterize it so that it _should_ work with other sovereign clouds as well.  @delbert I have also had integrated vectorization working in gov, if you want to reach out on Teams, I can walk you through that.

**Comment by pamelafox** [comment]
@bcperry We'd love a PR, assuming the changes are compatible with all existing functionality.
